engine/src/main/java/org/terasology/rendering/nui/layouts/FlowLayout.java

```diff
@@ -72,10 +80,10 @@ public Vector2i getPreferredContentSize(Canvas canvas, Vector2i sizeHint) {
         int filledHeight = 0;
         for (UIWidget widget : contents) {
             Vector2i size = canvas.calculatePreferredSize(widget);
-            if (filledWidth != 0 && filledWidth + size.x  > sizeHint.x) {
+            if (filledWidth != 0 && filledWidth + size.x > sizeHint.x) {
                 result.x = Math.max(result.x, filledWidth);
                 result.y += filledHeight;
-                filledWidth = size.x;
+                filledWidth = size.x + horizontalSpacing * (contents.size() - 1);
```

Comment:
This worked at first with 2 test, but the rest tests proven for this not being the correct solution.

Reply:
I think the solution for this is similar to the vertical case - just add the `horizontalSpacing` each time when adding `size.x` somewhere.